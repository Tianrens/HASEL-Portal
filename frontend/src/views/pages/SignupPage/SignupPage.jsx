import { React, useState } from 'react';
import styles from './SignupPage.module.scss';
import { StyledButton } from '../../components/buttons/StyledButton';
import TextField from '../../components/TextField/CustomTextField';
import StyledHeader from '../../components/text/StyledHeader';
import HeroPageTemplate from '../../components/templates/HeroPageTemplate/HeroPageTemplate';
import { auth } from '../../../firebase';
import { signUpUser } from '../../../state/docs/userDoc';
import UserTypeDropdown from '../../components/TextField/UserTypeDropdown';

const SignupPage = () => {
    const [firstName, setFirstName] = useState('');
    const [lastName, setLastName] = useState('');
    const [upi, setUpi] = useState('');
    const [accountType, setAccountType] = useState('');

    const uoaEmail = auth?.currentUser?.email ?? 'Unknown email, please contact Admin';

    const handleSubmit = async (event) => {
        event.preventDefault();
        await signUpUser(firstName, lastName, upi, accountType);
    };

    return (
        <HeroPageTemplate>
            <div className={styles.content}>
                <h1 className={styles.header}>Finish setting up your HASEL Portal account.</h1>
                <div className={styles.accountDetailsContainer}>
                    <StyledHeader left>Account Details</StyledHeader>
                    <p className={styles.inputTitle}>UoA Email</p>
                    <p className={styles.uoaEmail}>{uoaEmail}</p>
                </div>

                <div>
                    <StyledHeader left>Personal Information</StyledHeader>
                    <form className={styles.form} autoComplete='off' onSubmit={handleSubmit}>
                        <div className={styles.nameContainer}>
                            <TextField
                                title='First Name'
                                placeholder='--'
                                className={styles.inputField}
                                defaultValue=''
                                setValue={setFirstName}
                            />
                            <TextField
                                title='Last Name'
                                placeholder='--'
                                className={styles.inputField}
                                defaultValue=''
                                setValue={setLastName}
                            />
                        </div>

                        <TextField
                            title='UPI'
                            placeholder='e.g. abur970'
                            className={styles.inputField}
                            defaultValue=''
                            setValue={setUpi}
                        />

                        <UserTypeDropdown
                            className={styles.inputField}
                            required
                            setValue={setAccountType}
                        />

                        <div className={styles.buttonContainer}>
                            <StyledButton type='submit'>Submit</StyledButton>
                        </div>
                    </form>
                </div>
            </div>
        </HeroPageTemplate>
    );
};

export default SignupPage;
