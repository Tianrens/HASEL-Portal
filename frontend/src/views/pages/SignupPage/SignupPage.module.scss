@import '../../../assets/colours';

.content {
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: center;
    padding: 0 10rem;
}

.header {
    color: $heading;
    margin-bottom: 2rem;
    text-align: center;
}

.accountDetailsContainer {
    margin-bottom: 1rem;
}

.uoaEmail {
    color: $body;
    font-weight: 500;
}

.inputTitle {
    color: $secondaryBlue;
    margin-bottom: 0.3rem;
}

.nameContainer { // TODO: use Alex's media queries for more of a responsive design
    display: flex;
    flex-direction: row;
}

.nameInput {
    width: 100%;
    &:first-child {
        margin-right: 1rem;
    }
}

.inputField {
    background-color: white;
}

.buttonContainer {
    display: flex;
    justify-content: center;
    margin-top: 2rem;
    margin-bottom: 2rem;
}
