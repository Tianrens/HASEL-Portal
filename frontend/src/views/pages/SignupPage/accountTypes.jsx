const accountTypes = ['Student', 'Staff', 'Academic', 'Other'];

export default accountTypes;
