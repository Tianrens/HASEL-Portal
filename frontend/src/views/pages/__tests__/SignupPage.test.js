import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import SignupPage from '../SignupPage/SignupPage';

test('Signup page renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <SignupPage />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
