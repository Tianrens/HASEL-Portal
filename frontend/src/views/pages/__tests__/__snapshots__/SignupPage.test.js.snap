// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup page renders properly 1`] = `
<div
  className="fullPage"
>
  <div
    className="heroPattern"
  >
    <div
      className="container"
    >
      <div
        className="content"
      >
        <h1
          className="header"
        >
          Finish setting up your HASEL Portal account.
        </h1>
        <div
          className="accountDetailsContainer"
        >
          <div
            className="container "
          >
            <h2
              className="text alignLeft "
            >
              <span>
                Account Details
              </span>
            </h2>
            <button
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root button undefined   small css-11qr2p8-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <span
                  aria-hidden={true}
                  className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
                >
                  logout
                </span>
              </span>
              Log Out
            </button>
          </div>
          <p
            className="inputTitle"
          >
            UoA Email
          </p>
          <p
            className="uoaEmail"
          >
            Unknown email, please contact Admin
          </p>
        </div>
        <div>
          <div
            className="container "
          >
            <h2
              className="text alignLeft "
            >
              <span>
                Personal Information
              </span>
            </h2>
          </div>
          <form
            autoComplete="off"
            className="form"
            onSubmit={[Function]}
          >
            <div
              className="nameContainer"
            >
              <div
                className="container"
              >
                <p
                  className="title"
                >
                  First Name
                  <span
                    className="asterisk"
                  >
                     *
                  </span>
                </p>
                <div
                  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root inputField css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      defaultValue=""
                      disabled={false}
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="--"
                      required={true}
                      type="text"
                    />
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-1a1ihhj"
                      >
                        <span
                          className="notranslate"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                className="container"
              >
                <p
                  className="title"
                >
                  Last Name
                  <span
                    className="asterisk"
                  >
                     *
                  </span>
                </p>
                <div
                  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root inputField css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      defaultValue=""
                      disabled={false}
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="--"
                      required={true}
                      type="text"
                    />
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-1a1ihhj"
                      >
                        <span
                          className="notranslate"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="container"
            >
              <p
                className="title"
              >
                UPI
                <span
                  className="asterisk"
                >
                   *
                </span>
              </p>
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root inputField css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    defaultValue=""
                    disabled={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="e.g. abur970"
                    required={true}
                    type="text"
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-1a1ihhj"
                    >
                      <span
                        className="notranslate"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="container"
            >
              <p
                className="title"
              >
                Account Type
                <span
                  className="asterisk"
                >
                   *
                </span>
              </p>
              <div
                className="MuiFormControl-root MuiTextField-root inputField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                title="Account Type"
              >
                <div
                  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <span
                      className="notranslate"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </div>
                  <input
                    aria-hidden={true}
                    autoFocus={false}
                    className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onChange={[Function]}
                    required={true}
                    tabIndex={-1}
                    value=""
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-1a1ihhj"
                    >
                      <span
                        className="notranslate"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="buttonContainer"
            >
              <button
                className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root button undefined     css-jh47zj-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="rightSide"
  >
    <div
      className="bgImg"
    />
  </div>
</div>
`;
